ci_framework/roles/cifmw_cephadm/tasks/bootstrap.yml

```diff
@@ -98,6 +98,18 @@
         {{ cifmw_cephadm_container_ns + '/' + cifmw_cephadm_container_image + ':' + cifmw_cephadm_container_tag|string }}
       changed_when: false
 
+    - name: Switch HA container images if a custom container registry is used
+      become: true
+      when:
+        - cifmw_cephadm_registry_url | length > 0
+        - cifmw_cephadm_container_ns != "quay.io/ceph"
+      ansible.builtin.shell: |
+        {{ cifmw_cephadm_ceph_cli }} config set mgr mgr \
+          mgr/cephadm/container_image_keepalived {{ cifmw_cephadm_container_ns + '/keepalived:latest' }}
+        {{ cifmw_cephadm_ceph_cli }} config set mgr mgr \
+          mgr/cephadm/container_image_haproxy {{ cifmw_cephadm_container_ns + '/haproxy:latest' }}
```

Comment:
TripleO used `{{ tripleo_cephadm_haproxy_container_image }}` and `{{ tripleo_cephadm_keepalived_container_image }}`. You could do that too (`s/tripleo_/cifmw_/g`) or I'm fine with what you have with `{{ cifmw_cephadm_container_ns + '/haproxy:latest' }}` and `{{ cifmw_cephadm_container_ns + '/keepalived:latest' }}` too.